protected/humhub/modules/user/models/fieldtype/BaseTypeVirtual.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+/**
+ * @link https://www.humhub.org/
+ * @copyright Copyright (c) 2020 HumHub GmbH & Co. KG
+ * @license https://www.humhub.com/licences
+ */
+
+namespace humhub\modules\user\models\fieldtype;
+
+/**
+ * Base type for virtual profile fields
+ *
+ * Virtual profile fields are read-only and can be used to display content
+ * from other sources (e.g. user table).
+ *
+ * @since 1.6
+ */
+abstract class BaseTypeVirtual extends BaseType
+{
+
+    /**
+     * @inheritdoc
+     */
+    public $isVirtual = true;
+
+    /**
+     * @inheritdoc
+     */
+    public function rules()
+    {
+        return [];
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function getUserValue($user, $raw = true)
```

Comment:
Why not abstract?

Reply:
Ok, unfortunately the parent function is not abstract. Maybe we could use a template method to force subclasses to implement a value getter e.g. `abstract getVirtualUserValue()`

Reply:
Good idea!